enterprise/api_client/sso_orchestrator.py

```diff
@@ -0,0 +1,140 @@
+            'enterprise-customer-sso-configuration-orchestration-complete',
+            kwargs={'configuration_uuid': config_pk},
+        )
+        return f"{lms_base_url}{path}"
+
+    def _get_orchestrator_configure_url(self):
+        """
+        get the configure url for the SSO Orchestrator API
+        """
+        # probably want config value validated for this
+        return urljoin(self.base_url, get_sso_orchestrator_configure_path()) 
+
+    def _create_auth_header(self):
+        """
+        create the basic auth header for requests to the SSO Orchestrator API 
+        """
+        orchestrator_username = get_sso_orchestrator_basic_auth_username()
+        if not orchestrator_username:
+            raise ClientError(
+                "Failed to create SSO Orchestrator auth headers: username required.",
+                HTTPStatus.INTERNAL_SERVER_ERROR.value
+            )
+        orchestrator_password = get_sso_orchestrator_basic_auth_password()
+        if not orchestrator_password:
+            raise ClientError(
+                "Failed to create SSO Orchestrator auth headers: password required.",
+                HTTPStatus.INTERNAL_SERVER_ERROR.value
+            )
+        return f"Basic {base64.b64encode(f'{orchestrator_username}:{orchestrator_password}'.encode('utf-8')).decode()}"
+
+    def _create_session(self):
+        """
+        create a requests session object
+        """
+        if not self.session:
+            self.session = requests.Session()
+            self.session.headers.update({
+                "Authorization": self._create_auth_header(),
+            })
+
+    # No real need for this method at the moment as the orchestrator only supports post
+    # to /configure
+    # def _get(self, url, params=None):
+    #     """
+    #     make a GET request to the SSO Orchestrator API
+    #     """
+    #     self._create_session()
+    #     response = self.session.get(url, params=params)
+    #     if response.status_code != HTTPStatus.OK.value:
+    #         raise ClientError(
+    #             f"Failed to get data from {url}: {response.text}",
+    #             response.status_code
+    #         )
+    #     return response.json()
+
+    def _post(self, url, data=None):
+        """
+        make a GET request to the SSO Orchestrator API
+        """
+        self._create_session()
+        response = self.session.post(url, data=data)
+        if response.status_code != HTTPStatus.OK.value:
+            raise ClientError(
+                f"Failed to post data to {url}: {response.text}",
+                response.status_code
+            )
+        return response.status_code
+
+    def configure_sso_orchestration_record(
+        self,
+        config_data,
+        config_pk,
+        enterprise_data,
+        is_sap=False, 
+        updating_existing_record=False,
+    ):
+        """
+        configure an SSO orchestration record
+        """
+        # TODO: validate config_data
```

Comment:
what do you mean by validate in this context?

Reply:
uhm that was an old comment, I think validation should come from the model method, not the api client.